src/directory/mmap_directory.rs

```diff
@@ -31,12 +32,26 @@ pub(crate) fn make_io_err(msg: String) -> io::Error {
     io::Error::new(io::ErrorKind::Other, msg)
 }
 
+#[cfg(unix)]
 /// Returns `None` iff the file exists, can be read, but is empty (and hence
-/// cannot be mmapped)
-fn open_mmap(
+/// cannot be mmapped), this method also allows for optional [Advice] to be
+/// specified for the given workload.
+fn open_mmap_with_advice(
     full_path: &Path,
     madvice_opt: Option<Advice>,
```

Comment:
Shouldn't this be non optional?

Reply:
It could be however, I did not want to make this a breaking change so chose to leave it as optional.

Reply:
This is not a breaking change. That's not a public API.

Reply:
@PSeitz: Not necessarily. We still need one way or another to handle absence of Advice.
@ChillFish8 : No this is not public API.

Reply:
There's still `open_mmap` for an API without Advice